src/indexer/index_writer.rs

```diff
@@ -1049,4 +1099,151 @@ mod tests {
         assert_eq!(num_docs_containing("b"), 0);
         fail::cfg("RAMDirectory::atomic_write", "off").unwrap();
     }
+
+    #[test]
+    fn test_add_then_delete_all_documents() {
+        // debug using
+        // find the name of the most recently compiled executable
+        // ls -t target/debug/deps/ | rg ^tantivy | head -1
+        // rust-gdb $MOST_RECENT
+        // $gdb break src/indexer/stamper.rs:tantivy::indexer::stamper::Stamper::stamp
+        // $gdb run --test-threads 1 --test test_add_then_delete_all_documents
+        let mut schema_builder = schema::Schema::builder();
+        let text_field = schema_builder.add_text_field("text", schema::TEXT);
+        let index = Index::create_in_ram(schema_builder.build());
+        let reader = index
+            .reader_builder()
+            .reload_policy(ReloadPolicy::OnCommit)
```

Comment:
Let's make it `ReloadPolicy::Manual`. You do the reload anyway manually. `OnCommit` introduce a race condition we want to avoid in unit tests.